features/steps/validation_handling.py

```diff
@@ -241,7 +253,8 @@ def inner(context, **kwargs):
                     context.instances =  flatten_list_of_lists([fn(context, inst=inst, **kwargs) for inst in flatten_list_of_lists(context.instances)])
                 else:
                     context.instances = list(map(attrgetter('inst'), filter(lambda res: res.severity == OutcomeSeverity.PASS, itertools.chain.from_iterable(fn(context, inst=inst, **kwargs) for inst in context.instances))))
-                pass
+                    if 'its final' in context.step.name.lower() and 'segment_type' in fn_args:
```

Comment:
I'm worried about this. I really prefer not to have a tight coupling between the invocation in the decorator and the semantics of individual statements. This is really hard to maintain and understand.

The addition of `'its final'` should be handled in the step implementation function.

Reply:
Agree.
This has been resolved elsewhere and `its final` is no longer in the decorator.